phabricatoremails/model/persistence.py

```diff
@@ -0,0 +1,17 @@
+from sqlalchemy import Column, Integer, BigInteger
```

Comment:
As I understand, in general engineering, a "model" is a representation of some kind of data (e.g.: `Recipient`) in the domain (sending emails). There will be models that are used in the database (e.g. `QueryPosition`), and models that are only used in-memory (e.g.: `Revision`).

It sounds like SQLAlchemy's convention to to mutate the term "model" to only refer to SQLAlchemy models, and I think that's unfortunate.

I think you're right that it makes sense to put db models and in-memory models in different locations. I'll follow the convention, since "when in rome" :)

Reply:
> Note: based on this graphql sample project, it looks like making a models.py file (rather than models/__init__.py) is also acceptable.

Agreed, if you are going to put everything in a single file, `models.py` makes more sense.

>It sounds like SQLAlchemy's convention to to mutate the term "model" to only refer to SQLAlchemy models, and I think that's unfortunate.

This convention is used in many ORMs and web frameworks (mostly back-end frameworks). It's probably a relic from the MVC world, and does not really apply to frontend frameworks (more of what you are describing, i.e. in-memory). I have yet to run into a server-side application that does not follow this; It makes sense to me, but at the end of the day it is a convention.

Reply:
> It's probably a relic from the MVC world, and does not really apply to frontend frameworks (more of what you are describing, i.e. in-memory)

This is surprising to me - though backend web frameworks have a lot of backend models that are persisted, they _also_ have in-memory models (e.g.: parsing the `body` of a request and not directly persisting it, but using it for something). :thinking: